### PR TITLE
Improve cycling docs

### PR DESCRIPTION
**Pull request by MetRonnie**
Sibling to https://github.com/cylc/cylc-flow/pull/4554

Also addresses comment from Element:
> Graphviz comes with a conda install. But we should list it as a non-Python dependency for pip install 

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.

### CODE REVIEW

File: src/user-guide/writing-workflows/scheduling.rst [hunk 1]
```diff
@@ -325,8 +325,22 @@ inferred from context (rules below).
    except where specific tasks, if any, depend on :term:`clock triggers <clock
    trigger>`.
 
+
+Format 3: ``R[limit?]/[datetime]/[interval]``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (MetRonnie) [inline src/user-guide/writing-workflows/scheduling.rst:330]: It feels a bit awkward talking about format 3, then 4, then 1. But that's probably the most sensible order

**Review by hjoliver — APPROVED** [review]
Very nicely explained!

File: src/user-guide/writing-workflows/scheduling.rst [hunk 2]
```diff
@@ -325,8 +325,22 @@ inferred from context (rules below).
...
+
 The most common full form for recurrences is
-``R[limit?]/[datetime]/[interval]``. In Cylc this can be condensed to:
+``R[limit?]/[datetime]/[interval]``. This is format number 3 in the ISO 8601
+standard. The datetime specifies the start of the cycling sequence.
+For example, ``R3/2000-01-01T00Z/P2D`` means
```
Comment (wxtim) [inline src/user-guide/writing-workflows/scheduling.rst:335]: Consider adding plain English:

```suggestion
For example, ``R3/2000-01-01T00Z/P2D`` means

   "Run 3 times / starting at 2000-01-01T00Z (Midnight on Jan 1st 2000) / and repeat every 2 days"

which expands to:
```

File: src/user-guide/writing-workflows/scheduling.rst [hunk 3]
```diff
@@ -368,16 +382,20 @@ hours, so the inferred interval is 1 day (daily), ``P1D``.
...
-Another supported form of ISO8601 :term:`recurrence` is:
-``R[limit?]/[interval]/[datetime]``. This form uses the
-datetime as the end of the cycling sequence rather than the start.
-For example, ``R3/P5D/20140430T06`` means:
+
+Format 4: ``R[limit?]/[interval]/[datetime]``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Another supported recurrence form is:
+``R[limit?]/[interval]/[datetime]`` (format number 4 in the ISO 8601 standard).
```
Comment (wxtim) [inline src/user-guide/writing-workflows/scheduling.rst:390]: A second reason why the section order and the format number don't match to add to the note at the top of the Format3 section?

File: src/user-guide/writing-workflows/scheduling.rst [hunk 4]
```diff
@@ -393,12 +411,40 @@ So, for example, you can write:
...
+
+Format 1: ``R[limit?]/[datetime]/[datetime]``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A less common recurrence form is ``R[limit?]/[datetime]/[datetime]``
+(format number 1 in the ISO 8601 standard). This uses the difference between
+the first datetime and the second datetime to set the recurrence interval.
+The first datetime is the start point. For example,
+``R3/2020-07-10/2020-07-15`` means
+
```
Comment (wxtim) [inline src/user-guide/writing-workflows/scheduling.rst:427]: I think that you should write this in plain English too:

"Run three times / starting with the 10th July 2020 / and repeating on the 15th (and using the interval between the 10th and 15th of 5 days to calculate further recurrences)"

**Review by wxtim — APPROVED** [review]
I've added some suggestions - I've approved because I think that the changes are a good idea, but not worth blocking the PR for (especially the idea of turning examples into tables which would require a certain amount of effort).

I think that the English language translations could be useful to some readers.

File: src/user-guide/writing-workflows/scheduling.rst [hunk 5]
```diff
@@ -325,8 +325,22 @@ inferred from context (rules below).
...
+.. code-block:: none
+
+   2000-01-01T00Z
+   2000-01-03T00Z
+   2000-01-05T00Z
+
+In Cylc, this form can be condensed to:
 
 .. code-block:: sub
```
[the remark this answers is not in the document]
Comment (MetRonnie) [reply src/user-guide/writing-workflows/scheduling.rst:351]: I had a go, but the RTD theme does not seem suited to wide tables due to its relatively narrow content wrapper. Not ideal but I think the examples are clear enough that a table is not essential

[the remark this answers is not in the document]
Comment (wxtim) [reply src/user-guide/writing-workflows/scheduling.rst:354]: Fair - I had to insert a horrible bodge in metomi-mo (https://github.com/MetOffice/metomi-docs/blob/master/src/_static/theme_overrides.css) to make tables play well with rtd.
